mapper/orm-coordination-outbox-polling/src/main/java/org/hibernate/search/mapper/orm/coordination/outboxpolling/cfg/UuidDataType.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.orm.coordination.outboxpolling.cfg;
+
+import static org.hibernate.search.mapper.orm.coordination.outboxpolling.cfg.HibernateOrmMapperOutboxPollingSettings.Defaults.COORDINATION_ENTITY_MAPPING_OUTBOX_EVENT_UUID_DATA_TYPE;
+
+import java.lang.invoke.MethodHandles;
+
+import org.hibernate.dialect.Dialect;
+import org.hibernate.dialect.SQLServerDialect;
+import org.hibernate.search.mapper.orm.coordination.outboxpolling.logging.impl.Log;
+import org.hibernate.search.util.common.annotation.Incubating;
+import org.hibernate.search.util.common.logging.impl.LoggerFactory;
+
+@Incubating
+public enum UuidDataType {
+	/**
+	 * Is database specific. Some databases might not work correctly with a particular representation of UUID hence the
+	 * default provides the value against which particular database was tested. If the database is not on the testing list
+	 * the default would use a character representation as, characters should be overall supported by a database at least
+	 * as a type.
+	 * <p>
+	 * Note that in case of MS SQL database using {@link #CHAR} representation might lead to undesired behavior and
+	 * occurrence of deadlocks.
+	 */
+	DEFAULT( COORDINATION_ENTITY_MAPPING_OUTBOX_EVENT_UUID_DATA_TYPE ) {
+		@Override
+		public String type(Dialect dialect) {
+			// See HSEARCH-4749 why MSSQL is treated differently
+			if ( dialect instanceof SQLServerDialect ) {
+				return BINARY.type( dialect );
+			}
+			return CHAR.type( dialect );
+		}
+	},
+	/**
+	 *
+	 */
+	UUID( "uuid" ),
+
+	BINARY( "uuid-binary" ),
+
+	CHAR( "uuid-char" );
```

Comment:
Any reason not to call the `binary` and `char`? Not that I'm not that accustomed to the Search codebase so there might be a good reason.

Reply:
The idea was to match the types from ORM, so that it might look more familiar to the users. I'd probably say let's keep it like that for now and then let Yoann make a final call 😄 
Everything else makes sense - will make the adjustments, thanks!

Reply:
Either is fine to me, but please keep the name of enum values and the external representation consistent unless there's a good reason.

I.e. either `BINARY` with external representation `binary` OR `UUID_BINARY` with external representation `uuid-binary`.